src/components/Signup.js

```diff
@@ -0,0 +1,90 @@
+import React, { useContext } from "react";
+import { auth } from "../firebase";
+import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { UserContext } from "../App";
+import useInput from "./hooks/useInput";
+
+const Signup = () => {
+  const [username, setNewUsername, resetUsername] = useInput("");
+  const [email, setNewEmail, resetEmail] = useInput("");
+  const [password, setNewPassword, resetPassword] = useInput("");
+
+  const { setIsUserLoggedIn, setMsg } = useContext(UserContext);
+  const navigate = useNavigate();
+
+  const handleSignUp = async (e) => {
+    e.preventDefault();
+    if (!password || !email || !username) return;
+
+    try {
+      const newUser = await createUserWithEmailAndPassword(
+        auth,
+        email,
+        password
+      );
+      if (!newUser) return;
+      if (newUser) {
```

Comment:
Since you return if no newUser exists, this line should always resolve to true, therefore the if condition is redundant